### PR TITLE
[SPARK-49530][PYTHON] Support pie subplots in pyspark plotting

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Support pie subplots in pyspark plotting.

### Why are the changes needed?
API parity with pandas.DataFrame.plot.pie, see [here](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.pie.html)

### Does this PR introduce _any_ user-facing change?
Pie subplots are supported as shown below:

```py
>>> from datetime import datetime
>>> data = [
...             (3, 5, 20, datetime(2018, 1, 31)),
...             (2, 5, 42, datetime(2018, 2, 28)),
...             (3, 6, 28, datetime(2018, 3, 31)),
...             (9, 12, 62, datetime(2018, 4, 30)),
...         ]
>>> columns = ["sales", "signups", "visits", "date"]
>>> df = spark.createDataFrame(data, columns)
>>> fig = df.plot(kind="pie", x="date", subplots=True)
>>> fig.show()
```
newplot (2)

### How was this patch tested?
Unit tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: python/pyspark/sql/plot/core.py [hunk 1]
```diff
@@ -306,8 +305,8 @@ def pie(self, x: str, y: str, **kwargs: Any) -> "Figure":
         ----------
         x : str
             Name of column to be used as the category labels for the pie plot.
-        y : str
-            Name of the column to plot.
+        y : str, optional
+            Name of the column to plot. If not provided, `subplots=True` must be passed,
```
Comment (HyukjinKwon) [inline python/pyspark/sql/plot/core.py:309]:
```suggestion
            Name of the column to plot. If not provided, `subplots=True` must be passed at `kwargs`.
```

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/sql/plot/plotly.py [hunk 2]
```diff
@@ -48,13 +48,34 @@ def plot_pyspark(data: "DataFrame", kind: str, **kwargs: Any) -> "Figure":
...
-    # TODO(SPARK-49530): Support pie subplots with plotly backend
     from plotly import express
 
     pdf = PySparkPlotAccessor.plot_data_map["pie"](data)
     x = kwargs.pop("x", None)
     y = kwargs.pop("y", None)
-    fig = express.pie(pdf, values=y, names=x, **kwargs)
+    subplots = kwargs.pop("subplots", False)
+    if y is None and not subplots:
+        raise PySparkValueError(errorClass="UNSUPPORTED_PIE_PLOT_PARAM", messageParameters={})
```
Comment (tedyu) [inline python/pyspark/sql/plot/plotly.py:58]: It seems we can use default value of `True` for `subplots` in this case (`y is None`).

**Comment by xinrong-meng** [comment]
Thank you @HyukjinKwon !

File: python/pyspark/sql/plot/plotly.py (hunk 2 above)
Comment (xinrong-meng) [reply python/pyspark/sql/plot/plotly.py:58]: We are trying to reach parity with pandas [here](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.pie.html). Thanks for the feedback!

**Pull request closed without merging**